### PR TITLE
CLI: Tune plugin version for auth/secret mounts

### PR DESCRIPTION
**Pull request by tomhjp**
* Adds `-plugin-version` flag to `vault secrets tune` and `vault auth tune` commands
* Also adds `-plugin-version` flag to `vault secrets enable` which slipped unnoticed previously
* The `plugin_register_test.go` tests uncovered a bug in listing plugins. If you register both an auth and secrets "my-plugin"@v1.0.0, listing plugins would previously report two v1.0.0 plugins called "my-plugin" for both auth and secret types, giving 4 plugins instead of the correct 2.

I had to move the `PluginVersion` field from `MountInput` to the embedded `MountInputConfig` in order to support the way the `tune` commands gather their inputs. Normally, those two structs' fields would correspond to fields in `MountEntry` and its embedded `MountConfig` respectively, but I have currently left the `Version` field inside of `MountEntry` instead of moving it across to `MountConfig`. It seems to work fine, but I wonder if I've missed a reason that's a bad idea.

### CODE REVIEW

File: command/auth_enable.go [hunk 1]
```diff
@@ -307,6 +306,10 @@ func (c *AuthEnableCommand) Run(args []string) int {
 		if fl.Name == flagNameTokenType {
 			authOpts.Config.TokenType = c.flagTokenType
 		}
+
+		if fl.Name == "plugin-version" {
```
Comment (swenson) [inline command/auth_enable.go:310]: extract this out to a variable, `flagNamePluginVersion`, maybe, similar to `flagNameTokenType`?

**Review by swenson — APPROVED** [review]
LGTM, one small nit/suggestion

**Comment by swenson** [comment]
Love all of the testing here!

File: command/auth_enable.go (hunk 1 above)
Comment (tomhjp) [reply command/auth_enable.go]: Yeah good point 👍 it's duplicated a lot now.

**Review by tvoran — APPROVED** [review]
This seems to work for me! I'm assuming the `plugin-version` option will be added/changed for `plugin register` and `plugin deregister` in a separate PR?

I'm puzzled by the test-go failures though...

**Comment by tomhjp** [comment]
Thanks for the reviews, and thanks for providing an awesome base for those tests :)

The test failure was a silly copy-paste error, listing secret mounts instead of auth methods - should go green on this commit.

@tvoran On `plugin register` and `plugin deregister`, I wasn't sure about switching to `plugin_version` in the API and `-plugin-version` for flags, because it's more obvious from the context what the version refers to. However, while testing I have found it a bit of extra cognitive load to have to remember whether I should use `-version` or `-plugin-version` depending on whether I'm dealing with the plugin catalog. WDYT, shall we use `plugin_version` and `-plugin-version` for the catalog too?

**Comment by tvoran** [comment]
@tomhjp That's a fair point about `-version` being more obvious from the context in the plugin commands; I could see `plugin_` being extraneous there. I could go either way.

**Pull request merged**